### PR TITLE
Regex validation for codes

### PR DESCRIPTION
**Pull request by NeverResponse**
| Q               | A
| --------------- | ---
| Bug fix?        | no |
| New feature?    | yes |
| BC breaks?      | yes |
| Related tickets | - |
| License         | MIT |

### CODE REVIEW

File: src/Sylius/Bundle/AddressingBundle/Resources/config/validation/Province.xml [hunk 1]
```diff
@@ -34,7 +34,7 @@
             </constraint>
             <constraint name="Regex">
                 <option name="message">sylius.province.code.regex</option>
-                <option name="pattern">/[A-Z]{2}-[^W\d]{2,}/</option>
```
Comment (pjedrzejewski) [inline src/Sylius/Bundle/AddressingBundle/Resources/config/validation/Province.xml]: This one was correct.

Comment (michalmarcinkowski) [inline src/Sylius/Bundle/AddressingBundle/Resources/config/validation/Province.xml]: `XX-XX` - countryCode-provinceCode

Comment (pamil) [inline src/Sylius/Bundle/AddressingBundle/Resources/config/validation/Province.xml]: What about provinces identified by three or more letters? Shall we allow for numbers in province codes?

Comment (michalmarcinkowski) [inline src/Sylius/Bundle/AddressingBundle/Resources/config/validation/Province.xml]: We use https://en.wikipedia.org/wiki/ISO_3166-2 so it should always be 2 letter code

**Comment by pjedrzejewski** [comment]
Thank you Jasiek!

**Pull request merged**